src/UNL/UCBCN/ActiveRecord/Record.php

```diff
@@ -317,6 +326,39 @@ public static function getByAnyField($class, $field, $value, $whereAdd = '')
         return $record;
     }
 
+    public static function getManyByAnyField($class, $field, $value, $whereAdd = '') {
+        $records = array();
+        $record = new $class;
+
+        if (!empty($whereAdd)) {
+            $whereAdd = $whereAdd . ' AND ';
+        }
+
+        $mysqli = self::getDB();
+        $sql    = 'SELECT * FROM '
+                    . $record->getTable()
+                    . ' WHERE '
+                    . $whereAdd
+                    . $field . ' = "' . $mysqli->escape_string($value) . '"';
+        $result = $mysqli->query($sql);
+
+        if (false === $result) {
+            throw new Exception($mysqli->errno.':'.$mysqli->error, 500);
+        }
+
+        if ($result->num_rows == 0) {
+            return false;
+        }
+
+        while (!empty($row = $result->fetch_assoc())) {
+            $record = new $class;
```

Comment:
This can by dangerous.  I remember that, a long time ago, we were doing something similar and saw that server performance took a steep drop.  We discovered that it was because initializing a bunch of objects and iterating over them was taking a lot of memory.   For example, imagine loading up 10k records this way.  The List class allows us to initialize a record object as we iterate, and thus allows for PHP to auto-destory unused objects and thus keep memory at a minimum.